src/extension.ts

```diff
@@ -286,38 +268,47 @@ export async function activate(context: ExtensionContext) {
 		context.subscriptions.push(commands.registerCommand(commandId, commandHandler));
     }
 
-	let running: Disposable = undefined;
 	const enable: boolean = workspace.getConfiguration('inmanta').ls.enabled;
 
 	if (enable) {
-		running = await startServerAndClient();
+		await startServerAndClient();
 	}
 
-	function stopIfRunning() {
-		if (running !== undefined) {
-			running.dispose();
-			running = undefined;
-		}
-	}
-
-	function restartLS() {
+	async function restartLS() {
 		log("restarting Language Server");
-		stopIfRunning();
+		await stopServerAndClient();
 		startServerAndClient();
 	}
 
 	context.subscriptions.push(workspace.onDidChangeConfiguration(async e => {
 		if (e.affectsConfiguration('inmanta')) {
 			const enable: boolean = workspace.getConfiguration('inmanta').ls.enabled;
 
-			stopIfRunning();
+			await stopServerAndClient();
 
 			if (enable) {
-				running = await startServerAndClient();
+				await startServerAndClient();
 			}
 		}
 	}));
 
 	let exportToServerChannel: OutputChannel = null;
 	registerExportCommand();
 }
+
+async function stopServerAndClient() {
+	if (client) {
+		await client.stop();
+		client = undefined;
+	}
+	if(serverProcess){
+		serverProcess.kill();
+		serverProcess = undefined;
+	}
+}
+
+export async function deactivate(){
+	await stopServerAndClient();
+	// Return undefined because the cleanup process is done synchronously.
+	return undefined;
```

Comment:
I think you can also do `return stopServerAndClient();` and the framework will `await` it?
But this should also work as it is.

Reply:
True. I will fix it.